build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -974,4 +974,77 @@ export const handlers: Handler[] = [
 			}
 		},
 	},
+	{
+		name: "npm-package-json-clean-script",
+		match,
+		handler: (file, root) => {
+			// This rule enforces the "clean" script will delete all the build and test output
+			let json;
+
+			try {
+				json = JSON.parse(readFile(file));
+			} catch (err) {
+				return "Error parsing JSON file: " + file;
+			}
+
+			const scripts = json.scripts;
+			if (scripts === undefined) {
+				return undefined;
+			}
+
+			const missing = missingCleanDirectories(scripts);
+
+			if (missing.length !== 0) {
+				return `Clean script missing the following:${missing
+					.map((i) => `\n\t${i}`)
+					.join("")}`;
+			}
+		},
+		resolver: (file, root) => {
+			updatePackageJsonFile(path.dirname(file), (json) => {
+				const missing = missingCleanDirectories(json.scripts);
+				const clean = json.scripts["clean"] ?? "rimraf --glob";
+				if (missing.length === 0) {
+					return;
+				}
+				json.scripts["clean"] = `${clean} ${missing.map((name) => `'${name}'`).join(" ")}`;
+			});
+
+			return { resolved: true };
+		},
+	},
 ];
+
+function missingCleanDirectories(scripts: any) {
+	if (scripts.clean?.startsWith("pnpm")) {
```

Comment:
It isn't immediately clear to me what this early return case is for. Can we add a comment to it?

Reply:
Added comment and do additional check for "rimraf --glob" prefix.